aws/scripts/py/cache_invalidation.py

```
@@ -0,0 +1,302 @@
+            origin.get("DomainName", "")
+            for origin in distribution.get("Origins", {}).get("Items", [])
+        ]
+
+        # Skip app distributions
+        if self._is_app_distribution(origin_domains):
+            self.logger.info(f"Skipping app distribution: {dist_id}")
+            return None
+
+        # Classify as staging or production
+        if self._is_staging_distribution(distribution, origin_domains):
+            self.logger.info(f"Found staging distribution: {dist_id}")
```

Ruff (0.11.9)

Logging statement uses f-string

(G004)